### PR TITLE
[workload-testing] Misc improvements

### PR DESCRIPTION
**Pull request by radical**
-  **Use `PackageDownload` instead of `PackageReference`**
.. to get manifest, and workload pack nugets. The task does not need
these to be references as it essentially just copies files from the
restored location to the sdk. But using `PackageDownload` allows
packages with `PackageType=DotnetPlatform` to work too, which is most
manifest nugets.
- **Add property for output of workload-install command**
- **Add support for emitting packageSourceMappings in nuget.config**
- **Add PackageSourceNameForBuiltPackages property that defaults to nuget-local**

cc @joperezr

### CODE REVIEW

File: src/mono/nuget/Microsoft.NET.Runtime.WorkloadTesting.Internal/README.md [hunk 1]
```diff
@@ -5,11 +5,35 @@ Tasks, and targets to support workload testing in `dotnet` repositories.
 - `$(InstallWorkloadForTesting)` - required
 - `$(BuiltNuGetsDir)` - required
 - `$(DotNetInstallArgumentsForWorkloadsTesting)` - required
+
 - `$(TestUsingWorkloads)` - optional
 - `$(SkipTempDirectoryCleanup)` - optional
 - `$(VersionBandForManifestPackages)` - optional
 - `$(ExtraWorkloadInstallCommandArguments)` - optional
+- `$(WorkloadInstallCommandOutputImportance)` - optional
```
Comment (ilonatommy) [inline src/mono/nuget/Microsoft.NET.Runtime.WorkloadTesting.Internal/README.md:13]:
```suggestion
- `$(WorkloadInstallCommandOutputImportance)` - optional, defaults to `Normal`
```

File: src/tasks/WorkloadBuildTasks/PatchNuGetConfig.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        {
+            Log.LogError(laee.Message);
+            return false;
+        }
+    }
+
+    private static void Validate(string? templateNuGetConfigPath, string? packageSourceNameForBuiltPackages, string? outputPath)
+    {
+        if (string.IsNullOrEmpty(templateNuGetConfigPath))
+            throw new LogAsErrorException($"{nameof(templateNuGetConfigPath)} is required");
```
Comment (ilonatommy) [inline src/tasks/WorkloadBuildTasks/PatchNuGetConfig.cs:72]: In the doc we say `$(TemplateNuGetConfigPathForWorkloadTesting) - optional`. But here it's required. Am I missing something?

**Review by ilonatommy — APPROVED** [review]

File: src/tasks/WorkloadBuildTasks/PatchNuGetConfig.cs (hunk 2 above)
Comment (radical) [reply src/tasks/WorkloadBuildTasks/PatchNuGetConfig.cs:72]: Excellent catch!

Comment (radical) [reply src/tasks/WorkloadBuildTasks/PatchNuGetConfig.cs:72]: fixed.

**Review by maraf — APPROVED** [review]

**Pull request merged**